sdk/azidentity/environment_credential.go

```diff
@@ -9,30 +9,38 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 )
 
-// NewEnvironmentCredential creates an instance of the ClientSecretCredential type and reads credential details from environment variables.
+// NewEnvironmentCredential creates an instance that implements the azcore.TokenCredential interface and reads credential details from environment variables.
 // If the expected environment variables are not found at this time, then a CredentialUnavailableError will be returned.
 // options: The options used to configure the management of the requests sent to Azure Active Directory.
-func NewEnvironmentCredential(options *TokenCredentialOptions) (*ClientSecretCredential, error) {
+func NewEnvironmentCredential(options *TokenCredentialOptions) (azcore.TokenCredential, error) {
```

Comment:
Does this change need to be made elsewhere?  Also, do we need to continue to export ClientSecretCredential?

Reply:
All of our credential types are exported. Only NewEnvironmentCredential and NewDefaultAzureCredential arent types because they return other credentials. I can see a case for return azcore.TokenCredential from all of the constructors if we dont want to export any credential types.

Reply:
Having the constructors return interfaces gives us the ability to change the underlying implementation without affecting callers.  It also allows mocking, especially given that all these credential types don't export any content.

Reply:
Yeah I think that could provide us some flexibility, but I do think we need to get @JeffreyRichter to approve on this idea as well. @JeffreyRichter, what do you think of having all of the constructors return azcore.TokenCredential and unexporting the credential types?

Reply:
Thinking about this more, there are some advantages to returning `ClientSecretCredential`.  E.g. today it supports `azcore.TokenCredential`.  Say we end up with some new credential interface type in `azcore`.  If `ClientSecretCredential` can support both then this constructor continues to be useful for both cases.

The mocking of credentials can always be done for the API consuming the credential as it takes an interface, so I think that's less useful here.

Reply:
True, the current implementation is better if we ever want to check credential types with interfaces other than TokenCredential. But then, is it a good idea to return azcore.TokenCredential from NewEnvironmentCredential?

Reply:
You might want to leave it as `ClientSecretCredential` for now until we discuss further (can always switch to interfaces later).

Reply:
Hmmm, well the whole point of this change was to add support for ClientCertificateCredential and UsernamePasswordCredential from the env credential...I'll leave this PR open while we discuss

Reply:
Derp my bad, I overlooked that part.